libs/cdk/src/lib/template-management/template-manager.ts

```diff
@@ -0,0 +1,168 @@
+import { EmbeddedViewRef, TemplateRef } from '@angular/core';
+import { combineLatest, EMPTY, isObservable, Observable, of } from 'rxjs';
+import {
+  catchError,
+  distinctUntilChanged,
+  filter,
+  map,
+  merge as mergeWith,
+  switchAll,
+  switchMap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { onStrategy, rxMaterialize } from '../render-strategies/utils';
+import { RenderWork, RxNotification, RxNotificationKind } from '../render-strategies/model';
+import {
+  RenderSettings,
+  RxBaseTemplateNames,
+  RxViewContext,
+  TemplateSettings,
+} from './model';
+import {
+  getEmbeddedViewCreator,
+  getTNode,
+  notificationKindToViewContext,
+  notifyAllParentsIfNeeded,
+  notifyInjectingParentIfNeeded,
+  strategyHandling,
+  templateHandling,
+  templateTriggerHandling,
+  TNode,
+} from './utils';
+import { CoalescingOptions } from '../model';
+
+export interface RenderAware<T> {
+  nextStrategy: (nextConfig: string | Observable<string>) => void;
+  render: (values$: Observable<T>) => Observable<any>;
```

Comment:
```suggestion
  render: (values$: Observable<T>) => Observable<T | false | Error>;
```
hmm any seems kind of fishy... is this one correct?

Reply:
What this reminds me of is we are inconsistent across `TemplateManager` and `ListManager` in the render function output.

TemplateManager:
```typescript
export interface RenderAware<T> {
  nextStrategy: (nextConfig: string | Observable<string>) => void;
  render: (values$: Observable<T>) => Observable<any>;
}

export interface TemplateManager<
  T,
  C extends RxViewContext<T>,
  N = RxBaseTemplateNames | string
> extends RenderAware<T> {
  addTemplateRef: (name: N, templateRef: TemplateRef<C>) => void;
  addTrigger: (trigger$: Observable<RxNotification<T>>) => void;
}
```
vs
```typescript
export interface ListManager<T, C> {
  nextStrategy: (config: string | Observable<string>) => void;
  render(changes$: Observable<NgIterable<T>>): Observable<C>;
}
```

I like the version where we return the context. It contains all information possible:
- RxContext => suspense, next, error, complete
- additional context variables depending on the directive
-

Reply:
but `render(changes$: Observable<NgIterable<T>>): Observable<C>;` makes no sense at all.
`<C>` is the `RxViewContext` for a single item, but the renderCallback of `rxFor, (ListManager)` should return all items applied to the view? So just `NgIterable<T>` 

Otherwise we could think about emitting `<void>` as `renderCallback` as a starting point. I have used them a lot and never found myself in a situation where i needed the informationen passed via the callback.

Reply:
> <C> is the RxViewContext for a single item, but the renderCallback of rxFor
argh. right! Lets void here. agree